src/js/dp/search.js

```diff
@@ -18,31 +9,132 @@ const fetchHtml = async (url) => {
   return response && await response.text();
```

Comment:
We might want to catch any errors? Potentially retry with backoff? display a message. Probably need to speak to UX about what we'd then do